From 414762740301527852ddfaadcce5c5e29c35e42f Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Mon, 11 Nov 2019 11:56:37 -0800
Subject: [PATCH] Guard against option rom read.

Yes, this is overly paranoid. It is very likely that users may
"hit" the verify button without understanding what it's impact
may be. In most cases this shouldn't be needed anyway. The env
flag may be used to force it anyway.
---
 plugins/optionrom/fu-optionrom-device.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/plugins/optionrom/fu-optionrom-device.c b/plugins/optionrom/fu-optionrom-device.c
index 494fb00..36e83bb 100644
--- a/plugins/optionrom/fu-optionrom-device.c
+++ b/plugins/optionrom/fu-optionrom-device.c
@@ -46,6 +46,15 @@ fu_optionrom_device_read_firmware (FuDevice *device, GError **error)
 	g_autoptr(FuRom) rom = NULL;
 	g_autoptr(GBytes) fw = NULL;
 	g_autoptr(GFile) file = NULL;
+	const char *opt = getenv("FWUPD_READ_OPTION_ROM");
+
+	if ((!opt) || (strcmp(opt, "1") != 0)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "Reading option rom disabled: set FWUPD_READ_OPTION_ROM=1 in env first");
+		return NULL;
+	}
 
 	/* open the file */
 	rom_fn = g_build_filename (fu_udev_device_get_sysfs_path (udev_device), "rom", NULL);
-- 
2.24.0

